File: src/Applications/Orbital.Web/Orbital.Forecast.BlazorApplication/Orbital.Forecast.BlazorApplication.Client/Library/DynamicLinkLibraries/DataPerformer/DataPerformer.Interfaces/IDataRuntime.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace DataPerformer.Interfaces
{
    /// <summary>
    /// Runtime of data performer
    /// </summary>
    public interface IDataRuntime
    {

        /// <summary>
        /// Refreshes itself
        /// </summary>
        void Refresh();


        /// <summary>
        /// Starts all components
        /// </summary>
         /// <param name="time">Start time</param>
        void StartAll(double time);

        /// <summary>
        /// Updates all components
        /// </summary>
        void UpdateAll();

        /// <summary>
        /// Checks data consumer. Throws exceptions in case of fail
        /// </summary>
        /// <param name="dataConsumer">The data consumer</param>
        void Check(IDataConsumer dataConsumer);

        /// <summary>
        /// Gets differential solver from object
        /// </summary>
        /// <param name="obj">The object</param>
        /// <returns>The solver</returns>
        IDifferentialEquationSolver GetDifferentialEquationSolver(object obj);

        /// <summary>
        /// Time
        /// </summary>
        double Time
        {
            get;
            set;
        }

        /// <summary>
        /// Provider of time measure
        /// </summary>
        ITimeMeasurementProvider TimeProvider
        {
            get;
            set;
        }

        /// <summary>
        /// All components of runtime
        /// </summary>
        IEnumerable<object> AllComponents
        {
            get;
        }
    }
}

File: src/TypeScript/TypeScriptLibrary/Library/Measurements/Runtime/IDataRuntime.ts
import { ICategoryArrow } from "../../ICategoryArrow";
import { ICategoryObject } from "../../ICategoryObject";
import { ITimeMeasurementProvider } from "../ITimeMeasurementProvider";

export interface IDataRuntime {

    updateRuntime(): void;

    /// <summary>
    /// Refreshes itself
    /// </summary>
    refreshRuntime(): void;

    /// <summary>
    /// Starts all components
    /// </summary>
    /// <param name="time">Start time</param>
    startRuntime(time: number): void;


    setTimeProvider(timeProvider: ITimeMeasurementProvider): void;

    getTimeProvider(): ITimeMeasurementProvider;

    getRumtimeObjects(): ICategoryObject[];

    getRunimeArrows(): ICategoryArrow[];

}